src/app/appointments/AppointmentsList.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+import { useRouter } from "next/navigation";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Paper from "@mui/material/Paper";
+import { Box, Tooltip } from "@mui/material";
+import Container from "@mui/material/Container";
+import Sheet from "@mui/joy/Sheet";
+import EditCalendar from "@mui/icons-material/EditCalendar";
+import Delete from "@mui/icons-material/Delete"; // Uncomment if you have a delete icon
+
+function DateNewFormat(
+  dateString1: string | Date,
+  dateString2: string | Date
+): string {
+  let firstDate: Date;
+  if (typeof dateString1 === "string") {
+    firstDate = new Date(Date.parse(dateString1));
+  } else {
+    firstDate = dateString1;
+  }
+
+  let secondDate: Date;
+
+  if (typeof dateString2 === "string") {
+    secondDate = new Date(Date.parse(dateString2));
+  } else {
+    secondDate = dateString2;
+  }
```

Comment:
better initialize the variables with dateString1 and 2, and leave only the if if condition is met

```suggestion
  let firstDate: Date = dateString1;
  if (typeof dateString1 === "string") {
    firstDate = new Date(Date.parse(dateString1));
  } 
```